env0/data_teams.go

```diff
@@ -0,0 +1,48 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataTeams() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataTeamsRead,
+
+		Schema: map[string]*schema.Schema{
+			"names": {
+				Type:        schema.TypeList,
+				Description: "list of all teams (by name)",
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type:        schema.TypeString,
+					Description: "the team name",
+				},
+			},
+		},
+	}
+}
+
+func dataTeamsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+	teams, err := apiClient.Teams()
+	if err != nil {
+		return diag.Errorf("Could not get teams: %v", err)
+	}
+
+	data := []string{}
+
+	for _, team := range teams {
```

Comment:
I don't test if there are duplicate names. Wasn't sure it will be useful.

Reply:
Don't think it's relevant when reading from server